cupyx/scipy/sparse/compressed.py

```diff
@@ -546,41 +571,55 @@ def _get_single(self, major, minor):
                 data, indices, minor, answer)
         return answer[()]
 
-    def _get_major_slice(self, major):
-        major_size, minor_size = self._swap(*self._shape)
-        # major.indices cannot be used because scipy.sparse behaves differently
-        major_start = major.start
-        major_stop = major.stop
-        major_step = major.step
-        if major_start is None:
-            major_start = 0
-        if major_stop is None:
-            major_stop = major_size
-        if major_step is None:
-            major_step = 1
-        if major_start < 0:
-            major_start += major_size
-        if major_stop < 0:
-            major_stop += major_size
-        major_start = max(min(major_start, major_size), 0)
-        major_stop = max(min(major_stop, major_size), 0)
-
-        if major_step != 1:
-            raise ValueError('slicing with step != 1 not supported')
-
-        if not (major_start <= major_stop):
-            # will give an empty slice, but preserve shape on the other axis
-            major_start = major_stop
-
-        start = self.indptr[major_start]
-        stop = self.indptr[major_stop]
-        data = self.data[start:stop]
-        indptr = self.indptr[major_start:major_stop + 1] - start
-        indices = self.indices[start:stop]
-
-        shape = self._swap(len(indptr) - 1, minor_size)
-        return self.__class__(
-            (data, indices, indptr), shape=shape, dtype=self.dtype, copy=False)
+    def _get_submatrix(self, major=None, minor=None, copy=False):
```

Comment:
I don't want to allow a generic type for `major` and `minor` in this internal function for maintainability. How about converting to slice before each `_get_submatrix` call? I think it will be much clearer.